dex/pair/src/safe_price_view.rs

```diff
@@ -12,6 +13,12 @@ use crate::{
 pub const DEFAULT_SAFE_PRICE_ROUNDS_OFFSET: u64 = 10 * 60;
 pub const SECONDS_PER_ROUND: u64 = 6;
 
+#[derive(TypeAbi, TopEncode, TopDecode, NestedEncode, NestedDecode)]
+pub struct SafePriceLpToken<M: ManagedTypeApi> {
```

Comment:
Why use a struct instead of MultiValue2?

Reply:
I would not change the implementation here, as there may be a lot of external contracts that have implemented this feature as it is right now.

Reply:
Way easier to integrate into other contracts. You don't need to look into the source code to see what each multi-value means, you simply import the struct.